### DISCUSSION TITLE
Explore new pattern for one or multiple options

### DISCUSSION THREAD

**Discussion by afercia** [WordPress/gutenberg#42666] [General]
Splitting this out from https://github.com/WordPress/gutenberg/pull/42555 

In a few places in the Editor, a dropdown menu is used even when there's only one item in the dropdown. For example:

The template details popover:

<img width="550" alt="Screenshot 2022-07-20 at 12 45 00" src="https://user-images.githubusercontent.com/1682452/180725747-abeb8ccd-9e3b-41d0-8be7-ee8427e3a249.png">

The Templates list:

<img width="1165" alt="Screenshot 2022-07-20 at 12 44 30" src="https://user-images.githubusercontent.com/1682452/180725802-86180332-9205-41f4-930f-830340b2691f.png">

There's some agreement that this pattern is a bit frustrating, because it forces to unnecessarily click twice to access the only item available in the dropdown. Instead, when there's only one UI control, this should be immediately accessible to users and not buried down in a menu.

Generally, the. menu items are shown depending on some conditions. Actually, these menus may contain one or more items, depending on those conditions.

The `DropdownMenu` doesn't seem the most appropriate place to contain logic to count the number of items and show an alternative UI when there's only one item.

Possible options that come into my mind:
- Establish guidelines to not use a DropdownMenu in these cases. The Design should take into account the need for more space to display each control independently.
- Consider to make a new component that counts the number of children and shows:
  -  A simple Button when there's only one children.
  - A DropdownMenu when there's more than one children.
- Any better idea.

Any thoughts welcome.

Cc @jameskoster @javierarce @talldan

**Comment by jameskoster** [comment]
Possibly a terrible idea, but sharing to rule it out if nothing else :)

A spatially aware menu component, that utilises an ellipsis popover to contain the overflow:

<img width="1146" alt="Screenshot 2022-07-25 at 12 27 29" src="https://user-images.githubusercontent.com/846565/180767804-73cea980-0e31-44b6-8627-5e0741311d39.png">

Yes, sometimes there would be dropdowns with a single item, but the benefit is that other actions remain a single click away. This could potentially work well if the order has some logic (most common actions listed first).

**Reply by carolinan** [reply]
This is potentially confusing for users. There were a lot of support requests for the default theme Twenty Nineteen and it's so called "priority menu" because people did not understand why/that/where some menu items were suddenly hidden. Though that was for the themes primary menu.
